Source/CronosPlayUnreal/Private/DynamicContractActor.cpp

```diff
@@ -0,0 +1,241 @@
+
+void ADynamicContractActor::NewEthContract(FString cronosrpc,
+                                           FString contractaddress,
+                                           FString abijson, bool &success,
+                                           FString &output_message) {
+
+    try {
+        if (NULL != _coreContract) {
+            success = false;
+            output_message = TEXT("Contract Already Exists");
+            return;
+        }
+
+        std::string mycronosrpc = TCHAR_TO_UTF8(*cronosrpc);
+        std::string mycontract = TCHAR_TO_UTF8(*contractaddress);
+        std::string myjson = TCHAR_TO_UTF8(*abijson);
+
+        Box<EthContract> tmpContract =
+            new_eth_contract(mycronosrpc, mycontract, myjson);
+        // ownership transferred
+        _coreContract = tmpContract.into_raw();
+
+        success = true;
+    } catch (const rust::cxxbridge1::Error &e) {
+        success = false;
+        output_message =
+            FString::Printf(TEXT("CronosPlayUnreal NewEthContract Error: %s"),
+                            UTF8_TO_TCHAR(e.what()));
+    }
+}
+
+void ADynamicContractActor::NewSigningEthContract(
+    FString cronosrpc, FString contractaddress, FString abijson,
+    int32 walletindex, int64 chainid, bool &success, FString &output_message) {
+
+    try {
+        if (NULL != _coreContract) {
+            success = false;
+            output_message = TEXT("Contract Already Exists");
+            return;
+        }
+
+        if (NULL == defiWallet) {
+            success = false;
+            output_message = TEXT("Wallet Not Linked");
+            return;
+        }
+
+        assert(defiWallet != NULL);
+        if (NULL == defiWallet->_coreWallet) {
+            success = false;
+            output_message = TEXT("Wallet Not Initialized");
+            return;
+        }
+
+        char hdpath[100];
+        snprintf(hdpath, sizeof(hdpath), "m/44'/%d'/0'/0/%d", EthCoinType,
+                 walletindex);
+        rust::cxxbridge1::Box<PrivateKey> privatekey =
+            defiWallet->_coreWallet->get_key(hdpath);
+
+        std::string mycronosrpc = TCHAR_TO_UTF8(*cronosrpc);
+        std::string mycontract = TCHAR_TO_UTF8(*contractaddress);
+        std::string myjson = TCHAR_TO_UTF8(*abijson);
+
+        Box<EthContract> tmpContract = new_signing_eth_contract(
+            mycronosrpc, mycontract, myjson, *privatekey, chainid);
+        // ownership transferred
+        _coreContract = tmpContract.into_raw();
+
+        success = true;
+    } catch (const rust::cxxbridge1::Error &e) {
+        success = false;
+        output_message = FString::Printf(
+            TEXT("CronosPlayUnreal NewSigningEthContract Error: %s"),
+            UTF8_TO_TCHAR(e.what()));
+    }
+}
+
+void ADynamicContractActor::SendDynamicContract(
+    FString functionName, FString functionArgs,
+    FDynamicContractSendDelegate Out) {
+
+    AsyncTask(ENamedThreads::AnyHiPriThreadNormalTask, [this, Out, functionName,
```

Comment:
What if we call multiple times of SendDynamicContract? Any test results?

Reply:
i'll check,
i assume, if fired without waiting, second one will fail.